src/mlio/image_reader.cxx

```diff
@@ -0,0 +1,224 @@
+/*
```

Comment:
Can you add comments to make it easier to understand what each function is doing?

Reply:
I have tried to name each function in a way that makes its purpose clear. for example 'decode_image' is decoding the image and 'crop' is cropping the region of interest. Do you want more detailed comments regarding the logic within functions?

https://refactoring.guru/smells/comments
`The best comment is a good name for a method or class.`

Reply:
Yeah, I think in this case it would be helpful to have a few more detailed comments in the longer functions. Otherwise it's hard for a reader to understand conceptually how each function is being accomplished.